x-pack/metricbeat/module/aws/rds/rds.go

```diff
@@ -0,0 +1,269 @@
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(report mb.ReporterV2) error {
+	// Get startTime and endTime
+	startTime, endTime := aws.GetStartTimeEndTime(m.Period)
+
+	for _, regionName := range m.MetricSet.RegionsList {
+		awsConfig := m.MetricSet.AwsConfig.Copy()
+		awsConfig.Region = regionName
+		svc := rds.New(awsConfig)
+		// Get DBInstance ARNs per region
+		dbInstanceARNs, dbDetailsMap, err := getDBInstancesPerRegion(svc)
+		if err != nil {
+			err = errors.Wrap(err, "getDBInstancesPerRegion failed, skipping region "+regionName)
+			m.Logger().Errorf(err.Error())
+			report.Error(err)
+			continue
+		}
+
+		if len(dbInstanceARNs) == 0 {
+			continue
+		}
+
+		svcCloudwatch := cloudwatch.New(awsConfig)
+		namespace := "AWS/RDS"
+		listMetricsOutput, err := aws.GetListMetricsOutput(namespace, regionName, svcCloudwatch)
+		if err != nil {
+			m.Logger().Error(err.Error())
+			report.Error(err)
+			continue
+		}
+
+		if listMetricsOutput == nil || len(listMetricsOutput) == 0 {
+			continue
+		}
+
+		// Get MetricDataQuery for all dbInstances per region
+		var metricDataQueriesTotal []cloudwatch.MetricDataQuery
+		for _, dbInstanceARN := range dbInstanceARNs {
+			metricDataQueriesTotal = append(metricDataQueriesTotal, constructMetricQueries(listMetricsOutput, dbInstanceARN, m.Period)...)
+		}
+
+		var metricDataOutput []cloudwatch.MetricDataResult
+		if len(metricDataQueriesTotal) != 0 {
+			// Use metricDataQueries to make GetMetricData API calls
+			metricDataOutput, err = aws.GetMetricDataResults(metricDataQueriesTotal, svcCloudwatch, startTime, endTime)
+			if err != nil {
+				err = errors.Wrap(err, "GetMetricDataResults failed, skipping region "+regionName)
+				m.Logger().Error(err.Error())
+				report.Error(err)
+				continue
+			}
+		}
+
+		// Create Cloudwatch Events for RDS
+		events, err := createCloudWatchEvents(metricDataOutput, regionName, dbDetailsMap)
+		if err != nil {
+			m.Logger().Error(err.Error())
+			report.Error(err)
+			continue
+		}
+
+		for _, event := range events {
+			if len(event.MetricSetFields) != 0 {
+				if reported := report.Event(event); !reported {
+					m.Logger().Debug("Fetch interrupted, failed to emit event")
+					return nil
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func getDBInstancesPerRegion(svc rdsiface.RDSAPI) ([]string, map[string]DBDetails, error) {
+	describeInstanceInput := &rds.DescribeDBInstancesInput{}
+	req := svc.DescribeDBInstancesRequest(describeInstanceInput)
+	output, err := req.Send()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "Error DescribeDBInstancesRequest")
+	}
+
+	var dbInstanceARNs []string
+	dbDetailsMap := map[string]DBDetails{}
+	for _, dbInstance := range output.DBInstances {
+		dbInstanceARNs = append(dbInstanceARNs, *dbInstance.DBInstanceArn)
+		dbDetails := DBDetails{
```

Comment:
I'm wondering, does RDS suppor tags? do you get them? It would be a nice addition (for a follow up PR)

Reply:
Unfortunately this api for RDS does not support tags. If we want tags, then we need to make the `list-tags-for-resource` api separately. That is the one we are using in cloudwatch api.